src/main/java/com/iexec/common/docker/client/DockerClientInstance.java

```diff
@@ -557,9 +552,8 @@ public HostConfig buildHostConfigFromRunRequest(DockerRunRequest dockerRunReques
             hostConfig.withBinds(Binds.fromPrimitive(
                     dockerRunRequest.getBinds().toArray(new String[0])));
         }
-        if (dockerRunRequest.getDevices() != null) {
-            hostConfig.withDevices(dockerRunRequest.getDevices());
-        }
+        List<Device> devices = dockerRunRequest.getDevices();
```

Comment:
Nice! What about applying the same idea to both `dockerRunRequest.getXxx()` above?